pennylane/transforms/qcut.py

```diff
@@ -505,3 +505,162 @@ def contract_tensors(
+    transformation.
+
+    Args:
+        results (array-like): A collection of execution results corresponding to the
+            expansion of circuit fragments in the communication graph over measurement and
+            preparation node configurations. These results are processed into tensors by this
+            function.
+        prepare_nodes (Sequence[Sequence[PrepareNode]]): a sequence of size
+            ``len(communication_graph.nodes)`` that determines the order of preparation indices in
+            each tensor
+        measure_nodes (Sequence[Sequence[MeasureNode]]): a sequence of size
+            ``len(communication_graph.nodes)`` that determines the order of measurement indices in
+            each tensor
+
+    Returns:
+        List[array-like]: the tensors for each circuit fragment in the communication graph
+    """
+    ctr = 0
+    tensors = []
+
+    for p, m in zip(prepare_nodes, measure_nodes):
+        n_prep = len(p)
+        n_meas = len(m)
+        n = n_prep + n_meas
+
+        dim = 4**n
+        results_slice = qml.math.stack(results[ctr : dim + ctr])
+
+        tensors.append(_process_tensor(results_slice, n_prep, n_meas))
+
+        ctr += dim
+
+    if len(results) != ctr:
+        raise ValueError(f"The results argument should be a flat list of length {ctr}")
+
+    return tensors
+
+
+def qcut_processing_fn(
+    results: Sequence[Sequence],
+    communication_graph: MultiDiGraph,
+    prepare_nodes: Sequence[Sequence[PrepareNode]],
+    measure_nodes: Sequence[Sequence[MeasureNode]],
+    use_opt_einsum: bool = False,
+):
+    """Processing function for the :func:`cut_circuit` transform.
+
+    .. note::
+
+        This function is designed for use as part of the circuit cutting workflow. Check out the
+        :doc:`transforms </code/qml_transforms>` page for more details.
+
+    Args:
+        results (Sequence[Sequence]): A collection of execution results corresponding to the
+            expansion of circuit fragments in the ``communication_graph`` over measurement and
+            preparation node configurations. These results are processed into tensors and then
+            contracted.
+        communication_graph (MultiDiGraph): the communication graph determining connectivity between
+            circuit fragments
+        prepare_nodes (Sequence[Sequence[PrepareNode]]): a sequence of size
+            ``len(communication_graph.nodes)`` that determines the order of preparation indices in
+            each tensor
+        measure_nodes (Sequence[Sequence[MeasureNode]]): a sequence of size
+            ``len(communication_graph.nodes)`` that determines the order of measurement indices in
+            each tensor
+        use_opt_einsum (bool): Determines whether to use the
+            `opt_einsum <https://dgasmith.github.io/opt_einsum/>`__ package. This package is useful
+            for tensor contractions of large networks but must be installed separately using, e.g.,
```

Comment:
Maybe specify why it is useful (performance speed up I think) e.g "This should be used for performant tensor contractions of large networks but must be installed separately ...."

Reply:
Fixed here: f5c604ad2c10c3cd44c0596760904c9df5449305